### PR TITLE
[RWRoute] When removing unused source SPI restore intra-site routing

### PR DESCRIPTION
**Pull request by eddieh-xlnx**
Removal from https://github.com/Xilinx/RapidWright/pull/945

### CODE REVIEW

File: src/com/xilinx/rapidwright/rwroute/RWRoute.java [hunk 1]
```diff
@@ -1076,6 +1076,8 @@ protected void postRouteProcess() {
                 net.removePin(source, true);
                 assert(net.getSource() == altSource);
                 assert(net.getAlternateSource() == null);
+                // Restore the intra-site routing on the *_O site wire removed by removePin()
+                altSource.getSiteInst().routeIntraSiteNet(net, source.getBELPin(), source.getBELPin());
```
Comment (clavin-xlnx) [inline src/com/xilinx/rapidwright/rwroute/RWRoute.java:1080]: I think this makes sense for the case where `source==*_O` and `altSource==*MUX`, but what about the reverse? 

If `source==*MUX` and `altSource==*_O`, then aren't we are re-annotating the MUX output with the net?

Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/rwroute/RWRoute.java:1080]: After the call to `removePin()`, if we insert a `source = net.getSource()`, before this line, then I think it fixes the issue where its a problem in the first case and no-op in the second.

Comment (eddieh-xlnx) [reply src/com/xilinx/rapidwright/rwroute/RWRoute.java:1080]: Are you suggesting that we replace the diff with:
```
                source = altSource;
                si.routeIntraSiteNet(net, source.getBELPin(), source.getBELPin());
```
In this case, we'd be just routing the `*MUX` wire. The original intention was to route the `*_O` wire only. Alternatively, we'd have to compute the correct source BEL pin to reroute the entire path.

You do make a good point about the inverse case though. The simplest way would be to gate it with an `if` statement like the `if` block above.

**Review by clavin-xlnx — APPROVED** [review]

File: src/com/xilinx/rapidwright/rwroute/RWRoute.java (hunk 1 above)
Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/rwroute/RWRoute.java]: Sure, gating with `if` is the way to go.

**Pull request merged**